app/models/main.py

```diff
@@ -1653,13 +1656,37 @@ def validate(self, enforce_required=True, required_fields=None, max_day_rate=Non
         if errs:
             raise ValidationError(errs)
 
+    class query_class(BaseQuery):
+        def has_datetime_field_between(self, attr, start_datetime, end_datetime, inclusive=None):
```

Comment:
If `inclusive` is a bool, the default should be `False` rather than `None`? That said, if you are only ever calling with True, maybe the False path should just go away? (Actually, seems like what you really want in both cases is a half-closed range: it'd be better to add one to the day, rather than have to catch the last millisecond of the day in your criteria?

I'm not sure I quite like this abstraction (and if it does need to be a thing, it should be unit tested really). I wonder if it might be more clear to just add the filters you need in the two places.

Reply:
Some background: we introduced something similiar for the `Brief` model with the flexibility to filter by different date fields (`withdrawn_at`, `published_at` etc). This code is copied from there with a bit of tweaking. We were thinking of potentially DRYing up the shared stuff into a Mixin (I know how much we all love Mixins)?

Reply:
I just don't really see it as a particularly useful abstraction - it's inventing a new set of syntax & semantics for these particular cases, when in fact the task it's doing is not really anything special.

Reply:
Put it another way: it feels like if the function wasn't there, it'd be the same total volume of code? At which point I'm not clear there's a benefit to having it.

And yes having to pass in a field name feels rather like going against the way the ORM wants you to work, it's not the usual syntax/style as Rob says.